apps/cyberstorm-remix/cyberstorm/utils/StrongForm/useStrongForm.ts

```diff
@@ -57,20 +57,105 @@ export function useStrongForm<
   const [submitOutput, setSubmitOutput] = useState<SubmissionOutput>();
   const [submitError, setSubmitError] = useState<SubmissionError>();
   const [inputErrors, setInputErrors] = useState<InputErrors>();
+  const [fieldInteractions, setFieldInteractions] = useState<
+    Partial<
+      Record<
+        keyof Inputs,
+        {
+          hasFocused: boolean;
+          hasBlurred: boolean;
+        }
+      >
+    >
+  >({});
+  const [hasAttemptedSubmit, setHasAttemptedSubmit] = useState(false);
+
+  const isValueEmpty = (value: unknown) => {
```

Comment:
Nit: the body of useStrongForm is getting rather long, so a helper function like this could be moved elsewhere.